magento/magento2
# Fix loading of calendar script

Fix calendar mixins loading in Safari and Firefox.





### Description (*)
There is a bug with the loading of mixins in Safari and Firefox. 

When you call the widget using the imperative notation: `.calendar({`, the mixin for this module is not loading in Safari and Firefox. 

We change the calendar widget to be initialized in declarative notation to avoid this problem: `<script type="text/x-magento-init">`

### Manual testing scenarios (*)

1. Create a mixin for `mage/calendar`

app/code/Vendor/Module/view/frontend/requirejs-config.js:
```
var config = {
    config: {
        mixins: {
            'mage/calendar': {
                'Vendor_Module/js/lib/mage/calendar-mixin': true
            }
        }
    }
};
```
app/code/Vendor/Module/view/frontend/web/js/lib/mage/calendar-mixin.js:
```
define([
    'jquery'
], function ($) {
    'use strict';

    return function (widget) {
        $.widget('mage.calendar', widget.calendar, {
            _initPicker: function (element) {
               console.log('calendar mixin initialized');
           }
        });

        return $.mage.calendar;
    };
});
```

2. Open the page on the frontend with a date picker input (e.g. date of birth in customer account).

-- __Before Fix:__ --

**Expected result:** calendar-mixin.js is loading in Safari and Firefox (Dev Tools -> Network), `calendar mixin initialized` is in the console output.
**Actual result:** calendar-mixin.js is not loading in Safari and Firefox, no message in the console.

-- __After Fix:__ --

**Expected result:** calendar-mixin.js is loading in Safari and Firefox (Dev Tools -> Network), `calendar mixin initialized` is in the console output.
**Actual result:** calendar-mixin.js is loading in Safari and Firefox, `calendar mixin initialized` message written in the console.

### Related Pull Requests


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#32260: Fix loading of calendar script
